project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module

```diff
@@ -161,10 +161,214 @@ function c4m_og_menu() {
+ *   Form state array.
+ */
+function c4m_og_route_to_group_people($form, &$form_state) {
+  $gid = $form_state['values']['gid'];
+
+  drupal_goto('group/node/' . $gid . '/admin/people');
+}
+
+/**
+ * Function approves / rejects membership request, submitted by user.
+ *
+ * Verifies auto login token, and if valid, logs in the administrator and
+ * performs required state change to membership request. If not valid, forwards
+ * to login page.
+ *
+ * @param string $action
+ *   Approve / Reject.
+ * @param int $gid
+ *   Group ID.
+ * @param object $account
+ *   Administrator user that performs the action.
+ * @param object $membership
+ *   OG membership on which action is performed.
+ * @param int $timestamp
+ *   Timestamp of the moment action was requested.
+ * @param string $hashed_pass
+ *   Validation for executing user.
+ */
+function c4m_og_control_membership($action, $gid, $account, $membership, $timestamp = 0, $hashed_pass = '') {
+  global $user;
+  // Time out in seconds, until auto login token expires.
+  // 24 hours = 86400 seconds.
+  $timeout = 86400;
+  $current = REQUEST_TIME;
+
+  // If not logged in, or logged in as different user.
+  if (user_is_anonymous() || $account->uid != $user->uid) {
+    // Validate expiration and hashed password/login.
+    if ($timestamp <= $current && $current - $timestamp < $timeout && $account->uid && $timestamp >= $account->login &&
+      $hashed_pass == user_pass_rehash($account->pass, $timestamp, $account->login, $account->uid)) {
+
+      // If link data validated, perform automatic login.
+      $user = $account;
+      drupal_session_regenerate();
+    }
+    else {
+      // If not valid, forward to login page, with forwarding to membership
+      // edit page, after successful login.
+      drupal_set_message('Auto login token expired', 'warning');
+      $query = array('destination' => 'group/node/' . $gid . '/admin/people/edit-membership/' . $membership->id);
+      drupal_goto('user/login', array('query' => $query));
+    }
+  }
+
+  // Approve/Reject membership request.
+  if ($membership->state == OG_STATE_PENDING) {
+    $membership->state = $action == 'approve' ? OG_STATE_ACTIVE : OG_STATE_BLOCKED;
+    og_membership_save($membership);
+    $message = $action == 'approve' ? t('Membership approved!') : t('Membership rejected!');
+    drupal_set_message($message);
+  }
+  else {
+    drupal_set_message(t('Membership request already handled!'), 'warning');
+  }
+
+  if ($action == 'approve') {
+    // For 'reject' route will be performed by form['#submit'].
+    drupal_goto('group/node/' . $gid . '/admin/people');
+  }
+}
+
+/**
+ * Menu access; Check if entity is a group.
+ */
+function c4m_og_control_membership_access($group_type, $gid) {
+  $group = entity_load_single($group_type, $gid);
+  if (!$group || !og_is_group($group_type, $group)) {
+    return FALSE;
+  }
+
+  // Extract the group's bundle.
+  list(, , $bundle) = entity_extract_ids($group_type, $group);
```

Comment:
@ordavidil 
At og_role table, $group_type is 'node', and group_bundle are 'group/project'.